### PR TITLE
Various fixes for alignment problems

### PR DESCRIPTION
**Pull request by kholia**
These fixes should be safe.

See https://github.com/magnumripper/JohnTheRipper/issues/2868 and https://github.com/magnumripper/JohnTheRipper/pull/2911 for the discussion.

### CODE REVIEW

File: src/opencl_zip_fmt_plug.c [hunk 1]
```diff
@@ -389,7 +389,7 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	return count;
 }
 
-static int cmp_all(void *binary, int count)
+int cmp_all(void *binary, int count)
 {
```
Comment (jfoug) [inline src/opencl_zip_fmt_plug.c]: Why did you remove the static?  I think that is a problem.

**Comment by kholia** [comment]
Good catch @jfoug! I had removed `static` for debugging purposes. Fixed now.
